File: arrays/merge-overlapping-intervals.cpp
bool cmp(Interval a,Interval b){
    return a.start<b.start;
}
 
vector<Interval> Solution::merge(vector<Interval> &A) {
    int n = A.size();
    if(n==0) return A;
    sort(A.begin(),A.end(),cmp);
    
    vector<Interval> ans;
    ans.push_back(A[0]);
    
    for(int i=1; i<n; i++){
        if(A[i].start > ans[ans.size()-1].end) ans.push_back(A[i]);
        else if(A[i].end > ans[ans.size()-1].end)
        ans[ans.size()-1].end = A[i].end;
    }
    
    return ans;
}

File: arrays/merge-overlapping-intervals.py
# Definition for an interval.
# class Interval:
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution:
    def merge(self, intervals):
        intervals.sort(key = lambda x: x.start)

        i = 0
        while i < len(intervals)-1:
            first = intervals[i]
            second = intervals[i+1]

            if second.start <= first.end:
                first.end = max(first.end, second.end)
                del intervals[i+1]
            else:
                i += 1

        return intervals


